src/structured_data.rs

```diff
@@ -57,6 +57,19 @@ impl StructuredData {
                    previous_owner_signatures: previous_owner_signatures
                  }
     }
+
+    /// This is a static function required for computing a name give the tag-type and identifier to
+    /// be use by GETs
+    pub fn compute_name(type_tag: u64, identifier: &NameType) -> NameType {
```

Comment:
Explanation's in the documentaion :) .. Here it is in detail: In GETs I only have the type_tag and identifier, and i have to get the location by doing magic on it. That magic is the way structureddata name is calculated which also maintains an invariant. It needs to be available else i need to replicate the logic locally eveytime and it's not safe (if you ever decide to change the way name is calculated, none of my GETs will work)